tests/mix/testcrypto.nim

```diff
@@ -0,0 +1,210 @@
+{.used.}
+
+import chronicles, nimcrypto, results, unittest
+import ../../libp2p/protocols/mix/crypto
+
+suite "cryptographic_functions_tests":
+  test "aes_ctr_encrypt_decrypt":
+    let
+      key = cast[array[16, byte]]("thisis16byteskey")
+      iv = cast[array[16, byte]]("thisis16bytesiv!")
+      data: seq[byte] = cast[seq[byte]]("thisisdata")
+
+    let encryptedResult = aes_ctr(key, iv, data)
+    if encryptedResult.isErr:
+      error "Encryption failed", err = encryptedResult.error
+      fail()
+    let encrypted = encryptedResult.get()
+
+    let decryptedResult = aes_ctr(key, iv, encrypted)
+    if decryptedResult.isErr:
+      error "Decryption failed", err = decryptedResult.error
+      fail()
```

Comment:
hmm we could have this as utility-template, it actually seems vary interesting to be used in general. 
`require {condition}, {message}` 

in this case:
`require decryptedResult.isErr, "Decryption failed", decryptedResult.error`
`require {condition} {msg1} {msg2} .. {msgN}`

these tests here would greatly benefit as code would be much more smaller and readable. also we should adopt that in other tests in this repo.

Reply:
Ended up using `.expect()` and `check` since the former will concatenate the error message to the exception, and the latter since some of the messages are really not necessary, like:

```
    if emptyData != decrypted:
      error "Decrypted empty data does not match the original empty data",
        original = emptyData, decrypted = decrypted
      fail()
```

vs

```
check emptyData == decrypted
```

The test runner will inform us of that these variables dont match anyway.
This ended up reducing a lot of the test unit code